packages/fx-core/tests/component/generator/generator.test.ts

```diff
@@ -539,6 +539,25 @@ describe("Generator happy path", async () => {
     assert.isTrue(success);
     mockedEnvRestore();
   });
+
+  it("template from fallback", async () => {
+    const foobarTemplateName = "foobarTemplate";
+    const realTemplateName = "non-sso-tab";
+    const language = "ts";
+    const inputDir = path.join(tmpDir, "input");
+    await fs.ensureDir(path.join(inputDir, foobarTemplateName));
+    const fileData = "{{appName}}";
+    await fs.writeFile(path.join(inputDir, foobarTemplateName, "test.txt.tpl"), fileData);
+    const zip = new AdmZip();
+    zip.addLocalFolder(inputDir);
+    zip.writeZip(path.join(tmpDir, "foobar.zip"));
+    sandbox.stub(generatorUtils, "fetchTemplateZipUrl").resolves("foobar.zip");
+    sandbox
+      .stub(generatorUtils, "fetchZipFromUrl")
+      .resolves(new AdmZip(path.join(tmpDir, "foobar.zip")));
+    const result = await Generator.generateTemplate(context, tmpDir, realTemplateName, language);
+    assert.isTrue(result.isOk());
```

Comment:
Please also assert the results are expected.

Reply:
Thanks for the feedback, I'll assert more results.